### ISSUE TITLE
Can't assign 4 names to a 2 column data.table

### ISSUE THREAD

**Issue by NadineBestard** [SGDDNB/ShinyCell#12]
First of all I would like to thank you for this amazing package! doing a Shiny app with my single cell experiments had been for a while in my todo list and it is suddenley an easy task.

I successfully used the app with a seurat object. And now I tried to create another one from a single cell experiment object (sce).
```
> sce
class: SingleCellExperiment 
dim: 22681 18624 
metadata(1): Samples
assays(3): counts logcounts logcounts_raw
rownames(22681): Xkr4 Gm1992 ... CAAA01147332.1 AC149090.1
rowData names(6): ID Symbol ... gene_name subset
colnames(18624): 1_AAACCCAAGTATGGAT-1 1_AAACCCAGTTATGACC-1 ... 1_TTTGGTTGTATAGGGC-1 1_TTTGGTTTCCTCTTTC-1
colData names(27): Sample Barcode ... cluster_k60 celltype
reducedDimNames(3): PCA UMAP TSNE
altExpNames(0):
```

But I am getting the error bellow. 
![error_shiny](https://user-images.githubusercontent.com/61509376/118121711-c774bb00-b3e9-11eb-8182-34092d8edd38.PNG)

I tried to build the app convertring the same object to seurat. The conversion pops a few warning messages about changing some name labels (such as replacing the genenames with underscore with dashes), but then building the app works perfectly, therfore the object itself is not corrupted, it is just a format issue. 

I also tried to use the seurat object provided with the package transformed into a sce and that worked nicely. 
I tried to understand what is different from my sce but I don't know what is causing the error. 

PS: I aso went through an error along the lines of "can not find sampleID in sc1meta" that got solved by making sure I had adecuate colnames in my sce, just letting that here in case someone else gets that same error. 

I

**Comment by jfouyang** [comment]
Hi, thanks for using the package. 

First, let me summarise a few points so that we are on the same page:
- The example sce object provided with the package works well without any errors
- However, you get the above error when you use your own sce object
- Even after converting your own sce object into a Seurat object, the same error still persist

From the screenshot, it seems that the app has problems loading the Dimension Reduction X-axis and Y-axis. The two dropdown bars for Dimension Reduction X-axis and Y-axis is empty when they are supposed to show the possible Dimension Reduction e.g. UMAP1, UMAP2 etc. Can you help me check the following?

1. Can you run `head(reducedDim(sce, "UMAP"))` and show me what is inside? I want to check if the dimension reduction is stored properly.
2. In the shinyapp folder, there should be a file called "sc1conf.rds". Can you load the file in R and then run `sc1conf[grep("UMAP", ID)]`? I want to check if the dimension reduction information is available to shinycell.
3. Also, in the shinyapp folder, there should be a file called "sc1def.rds". Can you load the file in R and then run `sc1def$dimred`? I want to check what is the default dimension reduction.

**Comment by NadineBestard** [comment]
Hi, 
Let me clarify the points you summarise. 
- The example sce object provided with the package works well without any errors TRUE
- However, you get the above error when you use your own sce object TRUE
- Even after converting your own sce object into a Seurat object, the same error still persist  **FALSE** 
  - After converting the sce object to seurat the error disapears

1. Check the dimensional reduction is stored properly
The information is there however there are no colnames unlike for the PCs (that do appear in the dropdown menu)
```
> reducedDimNames(sce)
[1] "PCA"  "UMAP" "TSNE"

> head(reducedDim(sce, "UMAP"), 3)
                             [,1]       [,2]
1_AAACCCAAGTATGGAT-1  0.789401603   2.940407
1_AAACCCAGTTATGACC-1 -0.002099859 -13.584252
1_AAACCCATCATAGCAC-1  1.421826971   3.271134

> head(reducedDim(sce, "TSNE"), 3)
                          [,1]     [,2]
1_AAACCCAAGTATGGAT-1 -25.16637 20.72741
1_AAACCCAGTTATGACC-1  36.01388 21.03492
1_AAACCCATCATAGCAC-1 -20.16646 17.65342

> reducedDim(sce, "PCA")[1:4, 1:4]
                           PC1          PC2       PC3       PC4
1_AAACCCAAGTATGGAT-1  13.67322  -0.02909634 -6.059321 -2.867077
1_AAACCCAGTTATGACC-1 -15.63793 -27.55183191 -1.056772 -8.875582
1_AAACCCATCATAGCAC-1  13.14083   1.18697788 -6.057722 -1.052639
1_AAACCCATCCTACACC-1   6.26282  -4.76487182  5.032981 -1.838578

```

2. Is the dim reduction info available to the shiny cell? Does not look like it.
```
> sc1conf[grep("UMAP", ID)]
Empty data.table (0 rows and 7 cols): ID,UI,fID,fCL,fRow,grp...
```
3. Check what is the default dimension reduction
```
> sc1def$dimred
[1] "V1" "V2"
```

Thank you!

**Comment by jfouyang** [comment]
Hey, I finally found the problem! It seems like the lack of colnames in each of the reduced dimension matrices caused the problem. I implemented a fix that will check if the colnames are present and will add the colnames into ShinyCell if they are absent.

Can you try installing the latest version of ShinyCell and try it on your SCE object? Ideally, everything should work now.

**Comment by NadineBestard** [comment]
Thank you!
Now this works fine even when there are no column names in the dimensional reductions. 
I will close this issue as this is solved, but I will open a very similar one (#17), that arise  when colnames(sce) is NULL.
